front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { ComponentType, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { updateSpeedLimitByTag } from 'reducers/osrdconf';
+import { setFailure } from 'reducers/main';
+import icon from 'assets/pictures/components/speedometer.svg';
+import SelectImprovedSNCF from 'common/BootstrapSNCF/SelectImprovedSNCF';
+import { getInfraID, getSpeedLimitByTag } from 'reducers/osrdconf/selectors';
+import { Dispatch } from 'redux';
+import { noop } from 'lodash';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/query/react';
+import './SpeedLimitByTagSelector.scss';
+
+type SpeedLimitByTagSelectorProps = {
+  condensed?: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  speedLimitByTag?: string;
+  speedLimitsByTagsFromApi?: string[];
+  dispatchUpdateSpeedLimitByTag?: (tag: string) => Dispatch | void;
+};
+
+function withOSRDInfraData<T>(Component: ComponentType<T>) {
+  return (hocProps: T) => {
+    const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+    const dispatch = useDispatch();
+    const infraID = useSelector(getInfraID);
+    const speedLimitByTag = useSelector(getSpeedLimitByTag);
+    const { data, error } = osrdEditoastApi.useGetInfraByIdSpeedLimitTagsQuery(
+      {
+        id: infraID as number,
+      },
+      { skip: !infraID }
+    );
+    const dispatchUpdateSpeedLimitByTag = (newTag: string) => {
+      dispatch(updateSpeedLimitByTag(newTag));
+    };
+
+    useEffect(() => {
+      // Update the document title using the browser API
+      if (error) {
+        dispatch(
+          setFailure({
+            name: t('errorMessages.unableToRetrieveTags'),
+            message: `${(error as FetchBaseQueryError).status} : ${JSON.stringify(
+              (error as FetchBaseQueryError).data
+            )}`,
+          })
+        );
+      }
+    }, [error]);
+
+    return (
+      <Component
+        {...(hocProps as T)}
+        dispatch={dispatch}
+        speedLimitsByTagsFromApi={data}
+        speedLimitByTag={speedLimitByTag}
+        dispatchUpdateSpeedLimitByTag={dispatchUpdateSpeedLimitByTag}
+      />
+    );
+  };
+}
+
+export function IsolatedSpeedLimitByTagSelector({
+  speedLimitByTag,
+  condensed = false,
+  speedLimitsByTagsFromApi = [],
+  dispatchUpdateSpeedLimitByTag = noop,
+}: SpeedLimitByTagSelectorProps) {
+  const [speedLimitsTags, setSpeedLimitByTags] = useState<string[] | undefined>(
+    speedLimitsByTagsFromApi
+  );
+  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+
+  useEffect(() => {
+    // Update the document title using the browser API
+    setSpeedLimitByTags(speedLimitsByTagsFromApi);
+  }, [speedLimitsByTagsFromApi]);
+
+  return speedLimitsTags ? (
```

Comment:
```suggestion
  return speedLimitsTags.length > 0 ? (
```
By default speedLimitsTags will be [], so the current test is not very relevant (the component will be displayed even if speedLimitsTags is empty)

Reply:
Done